dms/models/ir_attachment.py

```diff
@@ -0,0 +1,61 @@
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from odoo import api, models
+
+
+class IrAttachment(models.Model):
+
+    _inherit = "ir.attachment"
+
+    def _get_dms_directories(self, res_model, res_id):
+        return self.env["dms.directory"].search(
+            [("res_model", "=", res_model), ("res_id", "=", res_id)]
+        )
+
+    def _dms_directories_create(self):
+        items = self._get_dms_directories(self.res_model, False)
+        for item in items:
+            model_item = self.env[self.res_model].sudo().browse(self.res_id)
+            ir_model_item = self.env["ir.model"].search(
+                [("model", "=", self.res_model)]
+            )
+            self.env["dms.directory"].create(
+                {
+                    "name": model_item.display_name,
+                    "model_id": ir_model_item.id,
```

Comment:
It was OK setting as well `res_model`. The problem you found is because the _compute_model_id is not completed right now.